### PR TITLE
Model selection in Step 5

### PR DESCRIPTION
**Pull request by J535D165**
This PR is a follow-up of PR #218 by @rvanelk. It implements the backend for the model selection.

### CODE REVIEW

File: asreview/webapp/utils/project.py [hunk 1]
```diff
@@ -60,13 +57,6 @@ def init_project(
                 'authors': project_authors
             }, fp)
 
-        asr_kwargs = deepcopy(app.config['asr_kwargs'])  # remove config
-        with open(get_kwargs_path(project_id), "w") as fp:
-            json.dump(asr_kwargs, fp)
-
-        # make a copy of the arguments to the state file
-        asr_kwargs['state_file'] = str(get_state_path(project_id))
```
Comment (J535D165) [inline asreview/webapp/utils/project.py:68]: @qubixes I'm a bit clueless on this line. Is this still required?

Comment (qubixes) [reply asreview/webapp/utils/project.py:68]: It seems not, so it should be safe to delete.

File: asreview/webapp/api.py [hunk 2]
```diff
@@ -392,6 +394,23 @@ def api_random_prior_papers(project_id):  # noqa: F401
...
+    ml_model = request.form.get("machine_learning_model", None)
+    if ml_model and not ml_model == "cli":
+        asr_kwargs["model"] = ml_model
+
+    # write the kwargs to a file
+    with open(get_kwargs_path(project_id), "w") as fp:
+        json.dump(asr_kwargs, fp)
+
+    # make a copy of the arguments to the state file
+    # asr_kwargs['state_file'] = str(get_state_path(project_id))  # TODO@{Raoul} obsolete?
```
Comment (qubixes) [inline asreview/webapp/api.py:412]: Yes, can be removed.

**Comment by qubixes** [comment]
Seems alright to me. Just one question; I don't see any mention of feature extraction selection. Not all combinations of feature extraction + model work well (i.e. they might use >10GB of memory), is something done to take care of that?

**Comment by J535D165** [comment]
At the moment, we do have 4 models implemented in the front-end. All with the default feature extraction method. I think the current combinations are fine, isn't it?

**Comment by qubixes** [comment]
Which 4? All except the Tensorflow models? That should work fine.

File: asreview/webapp/api.py [hunk 3]
```diff
@@ -392,6 +394,23 @@ def api_random_prior_papers(project_id):  # noqa: F401
...
     """Start training the model
     """
+
+    # get the CLI arguments
+    asr_kwargs = deepcopy(app.config['asr_kwargs'])
+
+    # add the machine learning model to the kwargs
+    # TODO@{Jonathan} validate model choice on server side
+    ml_model = request.form.get("machine_learning_model", None)
+    if ml_model and not ml_model == "cli":
```
Comment (qubixes) [inline asreview/webapp/api.py:404]: if ml_model != "cli"?

Isn't that equivalent/more concise?

File: asreview/webapp/utils/project.py (hunk 1 above)
Comment (J535D165) [reply asreview/webapp/utils/project.py:68]: Done

File: asreview/webapp/api.py (hunk 3 above)
Comment (J535D165) [reply asreview/webapp/api.py]: Thanks. I had a different intention with this part of the code. But I will propose an implementation later on. (It's not possible to experiment with new models now. So I wanted to a "CLI" model option in the front-end.)

**Comment by J535D165** [comment]
Thanks @rvanelk @qubixes @Rensvandeschoot for the work/feedback.

**Pull request merged**